rts/Rendering/Env/Particles/ProjectileDrawer.cpp

```diff
@@ -587,18 +594,18 @@ void CProjectileDrawer::DrawProjectilesSet(const std::vector<CProjectile*>& proj
 	}
 }
 
-bool CProjectileDrawer::CanDrawProjectile(const CProjectile* pro, const CSolidObject* owner)
+bool CProjectileDrawer::CanDrawProjectile(const CProjectile* pro, int allyTeam)
 {
 	auto& th = teamHandler;
 	auto& lh = losHandler;
-	return (gu->spectatingFullView || (owner != nullptr && th.Ally(owner->allyteam, gu->myAllyTeam)) || lh->InLos(pro, gu->myAllyTeam));
+	return (gu->spectatingFullView || (th.IsValidAllyTeam(allyTeam) && th.Ally(allyTeam, gu->myAllyTeam)) || lh->InLos(pro, gu->myAllyTeam));
```

Comment:
I think you made that change to enable `-1` as `allyTeam`, is there a particular reason you wanted to have it?